pipeline-adapters/mlrun-pipelines-kfp-v1-8/src/mlrun_pipelines/patcher.py

```diff
@@ -68,22 +68,29 @@ def _create_enriched_mlrun_workflow(
                 )
                 return workflow
 
-        # enrich each pipeline step with your desire k8s attribute
         for kfp_step_template in workflow["spec"]["templates"]:
-            if kfp_step_template.get("container"):
-                for function_obj in functions:
-                    # we condition within each function since the comparison between the function and
-                    # the kfp pod may change depending on the attribute type.
-                    _set_function_attribute_on_kfp_pod(
-                        kfp_step_template,
-                        function_obj,
-                        "PriorityClassName",
-                        "priority_class_name",
-                    )
-                    _enrich_kfp_pod_security_context(
-                        kfp_step_template,
-                        function_obj,
-                    )
+            if "dag" in kfp_step_template:
+                continue
+            env_vars = kfp_step_template["container"].get("env", [])
+            kfp_step_template["container"]["env"] = [
+                env_var
+                for env_var in env_vars
+                if env_var["name"] not in ["MLRUN_AUTH_SESSION", "V3IO_ACCESS_KEY"]
+            ]
```

Comment:
some comments would be nice for other developers trying to understand why you are filtering out those envvars

Reply:
Since this part is being done on the client side, I'm not sure how we would create the k8s secret here, as we have no k8s client initialized.